lstchain/tools/tests/test_tools.py

```diff
@@ -0,0 +1,100 @@
+import os
+import pytest
+import pandas as pd
+from ctapipe.core import run_tool
+
+
+@pytest.mark.run(after="test_lstchain_dl1_to_dl2")
+@pytest.mark.parametrize("point_like_IRF", [True, False])
+def test_create_irf(temp_dir_observed_files, simulated_dl2_file, point_like_IRF):
+    """
+    Generating an IRF file from a test DL2 files
+    """
+    from lstchain.tools.lstchain_create_irf_files import IRFFITSWriter
+    import json
+
+    irf_file = temp_dir_observed_files / "irf.fits.gz"
+    prov_log = temp_dir_observed_files / "temp_irffitswriter.provenance.log"
+    cuts = temp_dir_observed_files / "cuts.json"
+
+    if os.path.exists(cuts):
+        open(cuts, "r")
+    else:
+        data = json.load(open(os.path.join("lstchain/data/data_selection_cuts.json")))
+        data["fixed_cuts"]["gh_score"][0] = 0.3
+        data["events_filters"]["intensity"][0] = 0
+        json.dump(data, open(cuts, "x"))
+
+    assert (
+        run_tool(
+            IRFFITSWriter(),
+            argv=[
+                f"--input_gamma_dl2={simulated_dl2_file}",
+                f"--input_proton_dl2={simulated_dl2_file}",
+                f"--input_electron_dl2={simulated_dl2_file}",
+                f"--output_irf_file={irf_file}",
+                f"--point_like={point_like_IRF}",
+                f"--config_file={cuts}",
+            ],
+            cwd=temp_dir_observed_files,
+        )
+        == 0
+    )
+
+
+@pytest.mark.private_data
+@pytest.mark.run(after="test_create_irf")
+@pytest.mark.parametrize("add_IRF", [True, False])
+def test_create_dl3(temp_dir_observed_files, add_IRF):
+    """
+    Generating an DL3 file from a test DL2 files and test IRF file
+    """
+    from lstchain.tools.lstchain_create_dl3_file import DataReductionFITSWriter
+    from lstchain.io.io import dl2_params_lstcam_key
+    from lstchain.tests.test_lstchain import test_r0_path
+    import numpy as np
+
+    real_data_dl2_file = temp_dir_observed_files / (
+        "dl2_" + test_r0_path.with_suffix("").stem + ".h5"
+    )
```

Comment:
This file could be created as a fixture in `conftest` and then simply be parsed here as an input argument.

Reply:
True. It will be better that way.

Reply:
You could transform this test into a fixture in conftest:

https://github.com/cta-observatory/cta-lstchain/blob/365258afc85e779a77e9d993bda64a9327afd0f7/lstchain/scripts/tests/test_lstchain_scripts.py#L306

and then just test that the DL2 file produced is fine and use that fixture as input for the DL3 test

Reply:
I had a problem with just using this test into a fixture as the big `observed_dl1_files` is not a fixture and so it was not loading the dl2 file as a fixture to be used as an input for other tests.
I think as it is not extensively used, `real_data_dl2_file` can be just redefined as is, whenever needed.

Reply:
I see, then I think you also need to make sure that the observed dl2 file is produced beforehand with `@pytest.mark.run(after="test_lstchain_observed_dl1_to_dl2")` as well. It is in another module so I'm not sure if that marker makes sense though.

Reply:
No, I tried that in the `conftest.py` but it did not help, as it doesn't make any sense to use it 'after' the tests in `test_lstchain_scripts.py` are run